pkg/pillar/cmd/baseosmgr/handlebaseos.go

```diff
@@ -846,7 +849,37 @@ func validateBaseOsConfig(ctx *baseOsMgrContext, config types.BaseOsConfig) erro
 	return nil
 }
 
-func handleZbootTestComplete(ctx *baseOsMgrContext, config types.ZbootConfig, status types.ZbootStatus) {
+func completeUpgrade(version string) {
+	lastImageVersion := types.LastImageVersion{}
+	lastImageVersion.Version = version
+	lastImageVersion.LastUpgradeTime = time.Now()
+	data, err := json.Marshal(&lastImageVersion)
+	if err != nil {
+		log.Fatalf("***completeUpgrade: Failed to marshall "+
+			"LastImageVersion (%+v). err %s", lastImageVersion, err)
+	}
+	err = ioutil.WriteFile(LastImageVersionFile, data, 0644)
+	if err != nil {
+		log.Fatalf("***completeUpgrade: Failed to Save "+
+			"%s. err %s", LastImageVersionFile, err)
+	}
+	log.Debugf("completeUpgrade: Successfully saved %+v to %s",
+		lastImageVersion, LastImageVersionFile)
+
+	// run upgradeconverter
+	out, err := exec.Command("/opt/zededa/bin/upgradeconverter",
```

Comment:
You can't run this just because testing completed.
You can only run this when the user (through a TBD API) is telling the controller that the user has no desire to ever go back to an older version of eve. So please remove this code from baseosmgr.

Reply:
That isn't there currently - right? And this is the place where we finalize the Upgrade as it stands now.. When we add that confirmation from Controller - we need to move it there.